app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+  
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.